core/src/lib/cram_md5.cc

```diff
@@ -63,13 +67,26 @@ bool CramMd5Handshake::CramMd5Challenge()
   Mmsg(chal, "<%u.%u@%s>", (uint32_t)random(), (uint32_t)time(NULL),
        host.c_str());
 
-  Dmsg2(debuglevel_, "send: auth cram-md5 %s ssl=%d\n", chal.c_str(),
-        local_tls_policy_);
-  if (!bs_->fsend("auth cram-md5 %s ssl=%d\n", chal.c_str(),
-                  local_tls_policy_)) {
-    Dmsg1(debuglevel_, "Bnet send challenge comm error. ERR=%s\n",
-          bs_->bstrerror());
-    return false;
+  if (!bs_->IsBnetDumpEnabled()) {
```

Comment:
Consider turning this if/else around. I had to read it twice because I didn't notice the ! in the first place.
As a rule-of-thumb:
```
if (!condition) {
  whatever
}
```
is good practice until you add an else at the bottom.

Reply:
The statements in the if branch are the default behavior. However, I can invert the logic for readability.